streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -64,6 +64,7 @@
     private Cluster cluster;
     private Map<TaskId, Set<TopicPartition>> assignedActiveTasks;
     private Map<TaskId, Set<TopicPartition>> assignedStandbyTasks;
+    public boolean versionProbingFlag = false;
```

Comment:
Personally I think the version probing logic is exclusively owned by the streams partition assignor itself and leaking it into the task manager is a bit awkward... Note that right now this field is shared among StreamThread, TaskManager and StreamsPartitionAssignor.

But I can see that today we need to avoid TaskManager to change its internal tasks metadata when version probing happens so that we can still send the same subscription information again. Ideally we'd have our own `StreamsCoordinator` class than relying on `ConsumerCoordinator`. But for now, we can consider some alternative solution:

1. in StreamsConfig#InternalConfig, add another config like `__version.probing.flag__` to pass in a Boolean object, so that this object can be shared between the stream thread and streams partition assignor.
2. in StreamThread#listener#onPartitionAssigned, if boolean is true do not call taskManager#createTasks.

Although it is one more internal "backchannel" between these two classes, I feel it is actually cleaner than piggy backing on the task manager.

Reply:
Using `exception catch` rather than `a flag shared between the thread and the assignor` sounds a bit overkill to me, plus it is more costly: but arguably this situation should be rare so maybe it's fine..

I'm not very concerned about the checks: actually I think if we can avoid leaking it into `task manager`, the number of checks is actually quite small: one in the rebalance listener's `onPartitionAssigned`, and one in the thread's main loop.